pkg/planner/core/fragment.go

```diff
@@ -606,6 +610,10 @@ func (e *mppTaskGenerator) constructMPPTasksImpl(ctx context.Context, ts *Physic
 			SessionAlias:       sessionAlias,
 		}
 		tasks = append(tasks, task)
+		addr := meta.GetAddress()
+		if _, ok := e.nodeInfo[addr]; !ok {
+			e.nodeInfo[addr] = true
```

Comment:
Looks like just insert the addr to nodeInfo is ok? Why need to check it first?

Reply:
changed